### PR TITLE
Datalake UT

### PR DESCRIPTION
**Pull request by ashruti-msft**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,7 +1,11 @@
-## 2.2.1 (Unreleased)
+## 2.3.0 (Unreleased)
 **Bug Fixes**
 
 **Features**
+- Migrated to the latest [azblob SDK](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/storage/azblob).
+- Migrated to the latest [azfile SDK](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/storage/azfile).
+- Migrated from deprecated ADAL to MSAL through the latest [azidentity SDK](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azidentity).
+- Support for cold and premium tier
```
Comment (souravgupta-msft) [inline CHANGELOG.md:8]: We should elaborate this point. Tier change is not supported. We can upload a blob in the access tier specified in the config.
Also update the [base config file](https://github.com/Azure/azure-storage-fuse/blob/main/setup/baseConfig.yaml#L152).

**Review by vibhansa-msft — APPROVED** [review]

File: component/azstorage/azstorage.go [hunk 2]
```diff
@@ -323,6 +317,8 @@ func (az *AzStorage) StreamDir(options internal.StreamDirOptions) ([]*internal.O
 			options.Token = *new_marker
 			return az.StreamDir(options)
 		}
+	} else {
+		new_marker = to.Ptr("")
```
Comment (souravgupta-msft) [inline component/azstorage/azstorage.go:321]: This should have been updated right?

File: component/azstorage/datalake.go [hunk 3]
```diff
@@ -457,7 +457,6 @@ func (dl *Datalake) List(prefix string, marker *string, count int32) ([]*interna
 		}
 	}
 
-	// Process the paths returned in this result segment (if the segment is empty, the loop body won't execute)
```
Comment (souravgupta-msft) [inline component/azstorage/datalake.go:460]: Same as previous. Maybe sync this branch with the feature branch.

File: component/azstorage/datalake_test.go [hunk 4]
```diff
@@ -1164,40 +1168,39 @@ func (s *datalakeTestSuite) TestRenameFile() {
...
-	source := s.containerUrl.NewRootDirectoryURL().NewFileURL(src)
+	source := s.containerClient.NewFileClient(src)
 	s.az.CreateFile(internal.CreateFileOptions{Name: src})
 	// Add srcMeta to source
-	srcMeta := make(map[string]string)
-	srcMeta["foo"] = "bar"
-	source.CreateWithOptions(ctx, azbfs.CreateFileOptions{Metadata: srcMeta}, azbfs.BlobFSAccessControl{})
+	srcMeta := make(map[string]*string)
+	srcMeta["foo"] = to.Ptr("bar")
+	source.SetMetadata(ctx, srcMeta, nil)
```
Comment (souravgupta-msft) [inline component/azstorage/datalake_test.go:1189]: Should this call be Create or SetMetadata?

Comment (ashruti-msft) [reply component/azstorage/datalake_test.go:1189]: It should be the SetMetaData function only I think as the Create function creates the file taking createoptions as input not setting metadata

**Review by souravgupta-msft — APPROVED** [review]

**Pull request merged**